test/test_torch.py

```diff
@@ -17373,6 +17377,74 @@ def test_hypot(self, device, dtype):
                 expected = np.hypot(input[0].cpu().numpy(), input[1].cpu().numpy())
             self.assertEqual(actual, expected)
 
+    def _helper_test_igamma(self, loglo, loghi, device, dtype):
+        torch.manual_seed(123)
+
+        # generating random distribution from lo to hi with a loguniform
+        # distribution
+        def randrange(shape):
+            a = torch.rand(shape).to(device) * (loghi - loglo) + loglo
+            return torch.exp(a.to(torch.float64)).to(dtype)
+
+        vec1 = randrange((100, 1))
+        inputs = [
+            (randrange((10,)), randrange((10,))),
+            (randrange((3, 3, 3)), randrange((3, 3, 3))),
+            (randrange((100, 1)), randrange((100, 1)).transpose(0, 1)),
+            (vec1, vec1),  # for large number, it should approach 0.5
+            (vec1, 0.5 * vec1),  # test for considerable ratio
+            (vec1, 2.0 * vec1),
+        ]
+        half_prec = dtype in [torch.bfloat16, torch.float16]
+        for input0, input1 in inputs:
+            actual = torch.igamma(input0, input1)
+            if half_prec:
+                input0 = input0.to(torch.float)
+                input1 = input1.to(torch.float)
+            expected = scipy.special.gammainc(input0.cpu().numpy(), input1.cpu().numpy())
+            expected = torch.from_numpy(expected).to(dtype)
+            if half_prec:
```

Comment:
Is this check and its conversions necessary? I thought expected would be converted to the dtype of actual in the previous line.

Reply:
Ah, yes. It's probably from the older version of the commit that I forgot to change. Thanks!